packages/location-management/src/components/LocationForm/index.tsx

```diff
@@ -187,6 +190,7 @@ const LocationForm = (props: LocationFormProps) => {
 
           postPutLocationUnit(payload, openSRPBaseURL, service, isEditMode, params)
             .then(() => {
+              dispatch(fetchAllHierarchies([])); // reset tree data to force refresh of other component
```

Comment:
@ashfahan I think in other parts of code we are optimistically adding an item to the store when we add, or edit, shouldn't we do that here instead of resetting?

Reply:
the actual purpose of this pr was to change that implementation and instead of appending the data we use push the data so that we can control over what we are storing. And regarding the resetting of store in order to remove stale data from the store i am resetting the store value so that when it goes back into location unit view it will refetch the updated data from server with new changes

Reply:
@ashfahan I am with Kelvin on this one, I think we should rely more on optimistically updating the store; as for fetching data in other components, I do not think that its something that should be dependent on what data is in our store.

also; I think instead of passing a dispatch function here, you could instead add an `afterSubmitProp` that would get called after form is successfully submitted, this prop can contain any custom logic you would like to execute after form submission.

Reply:
done changing the dispatch to aftersubmit function